fedimint-api/src/db/mod.rs

```diff
@@ -695,38 +695,57 @@ where
 }
 
 /// This is a helper macro that generates the implementations of
-/// DatabaseKeyPrefixConst necessary for reading/writing to the
+/// [`DatabaseKeyPrefixConst`] necessary for reading/writing to the
 /// database and fetching by prefix.
 ///
 /// - `key`: This is the type of struct that will be used as the key into the
-///   database.
+///   database
 /// - `value`: This is the type of struct that will be used as the value into
-///   the database.
-/// - `prefix`: Enum expression that is represented as a `u8` that is the prefix
-///   prepended to this key.
-/// - `key_prefix`: Optional type of struct that can be passed multiple times if
-///   necessary. These will be used
-/// as the key prefix when querying the database via `find_by_prefix`.
+///   the database
+/// - `db_prefix`: Required enum expression that is represented as a `u8` and is
+///   prepended to this key
+/// - `query_prefix`: Optional type of struct that can be passed zero or more
+///   times. Every query prefix can be used to query the database via
+///   `find_by_prefix`
 ///
 /// Examples:
 ///
-/// `impl_db_prefix_const!(key = TestKey, value = TestVal, prefix =
-/// TestDbKeyPrefix::Test);`
+/// Only use the required parameters
 ///
-/// `impl_db_prefix_const!(key = AltTestKey, value = TestVal, prefix =
-/// TestDbKeyPrefix::AltTest, key_prefix = AltDbPrefixTestPrefix);`
+/// ```
+/// use crate::fedimint_api::impl_db_prefix_const;
+/// impl_db_prefix_const!(
+///     key = TestKey,
+///     value = TestVal,
+///     db_prefix =
+/// TestDbKeyPrefix::Test
```

Comment:
Was this automated by `cargo fmt`? Feels a little weird to have this on a newline

Reply:
yeah, good catch. it was definitely auto-formatted. perhaps because I didn't import the type.